src/logging.cpp

```diff
@@ -270,7 +353,7 @@ void BCLog::Logger::LogPrintStr(const std::string& str, const std::string& loggi
     for (const auto& cb : m_print_callbacks) {
         cb(str_prefixed);
     }
-    if (m_print_to_file) {
+    if (m_print_to_file && !skip_writing_to_disk_due_to_rate_limiting) {
```

Comment:
I'm not sure it's worth creating this variable `skip_writing_to_disk_due_to_rate_limiting`, maybe:
```
if (m_print_to_file && (skip_disk_usage_rate_limiting || !RateLimit(str_prefixed, logging_function, source_location))) {
```

Reply:
We want the rate limiting messages like "restarting logging from..." to also be printed to the console, so we have to call `RateLimit` before that and store the result. I am still gonna change the name of `skip_writing_to_disk_due_to_rate_limiting` because its a bit long.